README.md

```diff
@@ -133,10 +133,232 @@ Tests may be run individually (use `--help` for usage), or collectively via the
 ## Tests
 
 ### heap
+
+In x86 architecture, the stack grows up and towards the heap (which by
+convention grows in the opposite direction to the stack).
+![Heap and stack](heap-stack.png)
+
+By the sample `/proc/<pid>/maps` listing provided at the top, the stack is
+currently set at address range, 7fffd27c3000-7fffd27e400. Thus it only takes 47
+bits (log 7fffd27e4000 / log 2) to address this space. As the heap can't grow
+past the stack, this means that like the stack, only 47 bits are needed to address the heap
+even while using the 5-level page tables.
+
+This program tests to check if heap allocations can surpass 4-level page table tests.
+
 #### malloc
+
+The malloc test allocates as much memory as it can, 1 GiB at a time, pushing the
+heap up to higher and higher address ranges. (This memory is uninitialized.)
```

Comment:
wording suggestion:  This memory is left uninitialized.

I think the only reason it may be interesting to mention this fact is that if it were initialized, we'd actually need some storage for it... since we're only interested in using this test to determine addressability range, we don't touch it.  If we did, then the test would turn into a memory capacity test -- ie, how much memory can we fill on a given system.  The latter answer would be much, much smaller than the former.